ledger-service/http-json/src/main/scala/com/digitalasset/http/domain.scala

```diff
@@ -556,34 +573,50 @@ package domain {
     implicit val leftTraverseInstance: Traverse[ExerciseCommand[+*, Nothing]] =
       bitraverseInstance.leftTraverse
 
-    implicit val hasTemplateId =
+    implicit val hasTemplateId: HasTemplateId.Aux[ExerciseCommand[
+      +*,
+      domain.ContractLocator[_],
+    ], (Option[domain.ContractTypeId.Interface.Resolved], LfType)] =
       new HasTemplateId[ExerciseCommand[+*, domain.ContractLocator[_]]] {
 
         override def templateId(
             fab: ExerciseCommand[_, domain.ContractLocator[_]]
-        ): TemplateId.OptionalPkg = {
-          fab.reference match {
+        ): TemplateId.OptionalPkg =
+          fab.choiceInterfaceId getOrElse (fab.reference match {
             case EnrichedContractKey(templateId, _) => templateId
             case EnrichedContractId(Some(templateId), _) => templateId
             case EnrichedContractId(None, _) =>
               throw new IllegalArgumentException(
                 "Please specify templateId, optional templateId is not supported yet!"
               )
-          }
-        }
+          })
+
+        type TypeFromCtId = (Option[domain.ContractTypeId.Interface.Resolved], LfType)
 
         override def lfType(
             fa: ExerciseCommand[_, domain.ContractLocator[_]],
-            templateId: TemplateId.RequiredPkg,
+            templateId: ContractTypeId.Unknown.Resolved,
             f: PackageService.ResolveTemplateRecordType,
             g: PackageService.ResolveChoiceArgType,
             h: PackageService.ResolveKeyType,
-        ): Error \/ LfType =
+        ) =
```

Comment:
Why do we drop the type signature ?

Reply:
Specifying the type signature of an `override` is _usually_ pointless noise, and this change makes this one pointless and wrong noise. The only signal to be had is in the definition of `type TypeFromCtId` immediately above.

Reply:
OK.

Reply:
> Specifying the type signature of an `override` is _usually_ pointless noise, and this change makes this one pointless and wrong noise. The only signal to be had is in the definition of `type TypeFromCtId` immediately above.

I don't think it is. Having the signature fully available makes it easy to understand what the function does at a glance without having to go through the inheritance hiearachy.

Reply:
It's a typeclass and its instances. We don't restate the types of typeclass methods in every instance for the same reason.